File: 0383/main.swift
class Solution {
    func canConstruct(_ ransomNote: String, _ magazine: String) -> Bool {
        if ransomNote.count > magazine.count{
            return false
        }
        var hashmap = [Character: Int]()
        for value in magazine{
            hashmap[value] = (hashmap[value] ?? 0) + 1
        }
        for value in ransomNote{
            if let count = hashmap[value], count > 0{
                hashmap[value] = count - 1
            } else {
                return false
            }
        }
        return true
    }
}

File: 0383/main.py
from typing import Dict


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine):
            return False
        hashmap: Dict[str, int] = {}

        value: str = ""
        for value in magazine:
            hashmap[value] = hashmap.get(value, 0) + 1
        
        for value in ransomNote:
            if hashmap.get(value, 0) > 0:
                hashmap[value] -= 1
            else:
                return False
        
        return True
